src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -0,0 +1,1057 @@
+        final DetailNode[] children = javadoc.getChildren();
+        DetailNode javadocTagWithSince = null;
+        for (final DetailNode child : children) {
+            if (child.getType() == JavadocTokenTypes.JAVADOC_TAG) {
+                final DetailNode sinceNode = JavadocUtil.findFirstToken(
+                        child, JavadocTokenTypes.SINCE_LITERAL);
+                if (sinceNode != null) {
+                    javadocTagWithSince = child;
+                    break;
+                }
+            }
+        }
+        return javadocTagWithSince;
+    }
+
+    /**
+     * Get the default value of the property.
+     *
+     * @param propertyName the name of the property.
+     * @param field the field to get the default value of.
+     * @param classInstance the instance of the class to get the default value of.
+     * @return the default value of the property.
+     * @throws MacroExecutionException if an error occurs during getting the default value.
+     * @noinspection IfStatementWithTooManyBranches
+     * @noinspectionreason IfStatementWithTooManyBranches - complex nature of getting properties
+     *      from XML files requires giant if/else statement
+     */
+    // -@cs[CyclomaticComplexity] Splitting would not make the code more readable
+    private static String getDefaultValue(String propertyName, Field field, Object classInstance)
+            throws MacroExecutionException {
+        final Object value = SiteUtil.getFieldValue(field, classInstance);
+        final Class<?> fieldClass = getFieldClass(field);
+        String result = null;
+        if (CHARSET.equals(propertyName)) {
+            result = "the charset property of the parent Checker module";
+        }
+        else if ("PropertyCacheFile".equals(fieldClass.getSimpleName())) {
+            result = "null (no cache file)";
+        }
+        else if (fieldClass == boolean.class) {
+            result = value.toString();
+        }
+        else if (fieldClass == int.class) {
+            result = value.toString();
+        }
+        else if (fieldClass == int[].class) {
+            result = getIntArrayPropertyValue(value);
+        }
+        else if (fieldClass == double[].class) {
+            result = removeSquareBrackets(Arrays.toString((double[]) value).replace(".0", ""));
+            if (result.isEmpty()) {
+                result = CURLY_BRACKETS;
+            }
+        }
+        else if (fieldClass == String[].class) {
+            result = getStringArrayPropertyValue(propertyName, value);
+        }
+        else if (fieldClass == URI.class || fieldClass == String.class) {
+            if (value != null) {
+                result = '"' + value.toString() + '"';
+            }
+        }
+        else if (fieldClass == Pattern.class) {
+            if (value != null) {
+                result = '"' + value.toString().replace("\n", "\\n").replace("\t", "\\t")
+                        .replace("\r", "\\r").replace("\f", "\\f") + '"';
```

Comment:
Please share an example of a check that requires this replacement. Also, is it safe to just always assume that we need to do `\` -> `\\`?

Reply:
This method and others I have taken as-is from `XdocsPagesTest` - `getProperties`, `fixCapturedProperties`, `getType`, `getDefaultValue`, `getPatternArrayPropertyValue`, `getStringArrayPropertyValue`, `getIntArrayPropertyValue`, `getFieldClass`, `subtractTokens`.

To answer your question - running the code from `XdocsPagesTest`, `JavadocStyle.endOfSentenceFormat` pops up
![Screenshot from 2023-08-29 14-15-51](https://github.com/checkstyle/checkstyle/assets/23459549/117a5c2a-03b7-4db2-82e9-ae0a1e50fa70)


> Also, is it safe to just always assume that we need to do \ -> \\

I think so. Makes `value` readable:

Value before
```
([.?!][ 	

<])|([.?!]$)
```

Value after
```
([.?!][ \t\n\r\f<])|([.?!]$)
```